Server/VikingVault.Services/ExecutePaymentService.cs

```diff
@@ -22,44 +22,77 @@ public ExecutePaymentService(VikingVaultDbContext dbContext)
-
+                                                                        && bankAccount.CurrencyType.Equals("Ron"));
+               
                 if (payingUserAccount.Balance < automaticPayment.Amount)
                 {
-                    automaticPayment.IsEnabled = false;
-                    _dbContext.AutomaticPayments.Update(automaticPayment);
-                    _dbContext.SaveChanges();
+                    DisableAutomaticPayment(automaticPayment);
+                    AddFailedPaymentNotification(automaticPayment);
                 }
-                else
+                else if (!lastPaymentDate.Equals(currentDateTime.Date) && !payingUserCard.Blocked)
                 {
-                    if (!lastPaymentDate.Equals(currentDateTime.Date) && !payingUserCard.Blocked)
-                    {
-                        UpdateCompanyBankAccount(automaticPayment.ReceivingCompany, automaticPayment.Amount);
-                        UpdatePayingUserBankAccount(automaticPayment.PayingUser, automaticPayment.Amount);
-                        UpdateAutomaticPayment(automaticPayment, currentDateTime);
-                        AddPayingUserAutomaticTransaction(automaticPayment, currentDateTime);
-                    }
+                    UpdatePaymentInvolvedAccounts(automaticPayment, currentDateTime);
+                    AddPayingUserAutomaticTransaction(automaticPayment, currentDateTime);
+                    AddSuccessfulPaymentNotification(automaticPayment);
                 }
             }
         }
 
+        private IList<AutomaticPayment> CurrentPayments(DateTime currentDateTime)
+        {
+           return _dbContext.AutomaticPayments.Include(automaticPayment => automaticPayment.PayingUser)
+                                              .Include(automaticPayment => automaticPayment.ReceivingCompany)
+                                              .Where(automaticPayment => IsPaymentEligible(automaticPayment, currentDateTime))
+                                              .ToList();
+        }
+
+        private bool IsPaymentEligible(AutomaticPayment automaticPayment, DateTime currentDateTime)
+        {
+            return automaticPayment.InitialPaymentDate.Day.Equals(currentDateTime.Day) && automaticPayment.IsEnabled;
+        }
+
+        private void DisableAutomaticPayment(AutomaticPayment automaticPayment)
+        {
+            automaticPayment.IsEnabled = false;
+            _dbContext.AutomaticPayments.Update(automaticPayment);
+            _dbContext.SaveChanges();
+        }
+
+        private void AddSuccessfulPaymentNotification(AutomaticPayment automaticPayment)
+        {
+            Notification notification = new Notification
+            {
+                User = automaticPayment.PayingUser,
+                Text = "Payment to " + automaticPayment.ReceivingCompany.FirstName +
```

Comment:
I think this will look better using string interpolation `$"Basic chars here {awesomeInterpolatedVariableHere}"`